src/java.desktop/windows/native/libawt/windows/awt_ScrollPane.cpp

```diff
@@ -701,7 +701,7 @@ Java_sun_awt_windows_WScrollPanePeer_getOffset(JNIEnv *env, jobject self,
     gos->scrollpane = env->NewGlobalRef(self);
     gos->orient = orient;
 
-    return static_cast<jint>(reinterpret_cast<INT_PTR>(AwtToolkit::GetInstance().SyncCall(
+    return static_cast<jint>(reinterpret_cast<INT_PTR>(AwtToolkit::GetInstance().InvokeFunction(
```

Comment:
So this needs to block until the toolkit thread can process this and return, but since its
directly called from Java (ie we are in a JNI method) I think this is likely fine.

Reply:
That's right.

In addition to that, `WScrollPanePeer.getScrollOffset` is unused, so it's never called.

Reply:
If it is unused can we delete it?

Reply:
> If it is unused can we delete it?

It's possible. Yet I'd rather leave it for another time.

There's also a specific `WM_AWT_SET_SCROLL_INFO` message to call `::SetScrollInfo`. This message is never used, it can also be removed.